tools/codegen/gen_backend_stubs.py

```diff
@@ -0,0 +1,114 @@
+import pathlib
+import argparse
+import os
+import yaml
+from typing import List, Dict, Union, Tuple, Sequence
+from tools.codegen.gen import FileManager, get_grouped_native_functions, LineLoader, parse_native_yaml
+from tools.codegen.model import (ExternalBackendFunction, ExternalBackendFunctionsGroup,
+                                 NativeFunction, NativeFunctionsGroup, OperatorName,
+                                 ExternalBackendMetadata, assert_never)
+from tools.codegen.selective_build.selector import SelectiveBuilder
+from tools.codegen.utils import Target, concatMap
+import tools.codegen.dest as dest
+
+def parse_backend_yaml(
+        backend_yaml_path: str,
+        grouped_native_functions: Sequence[Union[NativeFunction, NativeFunctionsGroup]]
+) -> Tuple[str, List[Union[ExternalBackendFunction, ExternalBackendFunctionsGroup]]]:
+    with open(backend_yaml_path, 'r') as f:
+        yaml_values = yaml.load(f, Loader=LineLoader)
+    assert isinstance(yaml_values, dict)
+
+    cpp_namespace = yaml_values['cpp_namespace']
+
+    backend = yaml_values['backend']
+    supported = yaml_values['supported']
+    supported_autograd = yaml_values['autograd']
```

Comment:
As this YAML is user provided it is worth being more paranoid about the inputs provided here. Good input validation here will help save you from weird error messages later. `NativeFunction.from_yaml` does a pretty good job of validating its arguments and giving useful error information when things have gone wrong, and I recommend modeling the code here after it, if you don't have ideas on your own.

Reply:
You're right, thanks. I'm going to make the error'ing a little more robust in this PR. I'll probably add some explicit tests in a followup.